### PR TITLE
Feature/standlone linux tests

### LINKED ISSUE #150: Add test targets to standalone build
It would be nice if the library would come with the possibility to run our tests (on linux).

Below is a preliminary example of how to add a test to `standalone/CMakeLists.txt` which builds both with pure cmake and catkin.

However, this does not integrate well with the overall catkin project, since `catkin_add_gtest` does a bit more than just building an executable with gtest. (https://github.com/ros/catkin/blob/203a57ad70430ae165ebe014738a37c3af3e9166/cmake/test/gtest.cmake#L93)

```
if (NOT GTEST_FOUND)
  add_subdirectory("/usr/src/gtest" ${CMAKE_BINARY_DIR}/gtest)
endif()

enable_testing()

include_directories(
  test/include
  ${GTEST_INCLUDE_DIRS}
)

ADD_EXECUTABLE(unittest_laserscan_conversions test/unit_tests/unittest_laserscan_conversions.cpp)

TARGET_LINK_LIBRARIES(unittest_laserscan_conversions
    ${PROJECT_NAME}
    gtest
)

ADD_TEST(NAME unittest_laserscan_conversions
         COMMAND unittest_laserscan_conversions)
```

### PR DESCRIPTION
**Pull request by agutenkunst**
## Description

Also run tests on the standlone linux build. Needs #160. Close #150

This is a preparation for a following PR using this the also run the tests on the linux standalone job.

### Things to add, update or check by the maintainers before this PR can be merged.

* [x] ~Public api function documentation~
* [x] ~Architecture documentation reflects actual code structure~
* [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
* [x] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
* [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
* [ ] ~Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))~
* [x] ~CHANGELOG.rst updated~
* [x] ~Copyright headers~
* [x] ~Examples~

### Review Checklist
* [x] ~Soft- and hardware architecture (diagrams and description)~
* [x] ~Test review (test plan and individual test cases)~
* [x] Documentation describes purpose of file(s) and responsibilities
* [x] Code (coding rules, style guide)

### Release planning (please answer)
* [x] ~When is the new feature released?~
* [x] ~Which dependent packages do have to be released simultaneously?~

### CODE REVIEW

File: standalone/CMakeLists.txt [hunk 1]
```diff
@@ -84,7 +84,31 @@ if (WIN32)
 include(CTest)
```
Comment (agutenkunst) [inline standalone/CMakeLists.txt:84]: Ok this should be inside `BUILD_TESTING`

File: standalone/CMakeLists.txt.in [hunk 2]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
+
+project(googletest-download NONE)
+
+include(ExternalProject)
+ExternalProject_Add(googletest
+  GIT_REPOSITORY    https://github.com/google/googletest.git
+  GIT_TAG           master
```
Comment (agutenkunst) [inline standalone/CMakeLists.txt.in:8]: We could/should maybe fix this.

File: standalone/CMakeLists.txt [hunk 3]
```diff
@@ -80,10 +80,14 @@ target_link_libraries(${PROJECT_NAME}_app
...
-if (WIN32)
+if (BUILD_TESTING)
+
 include(CTest)
 
-if (BUILD_TESTING)
+# The standalone is included  with add_subdirectory from the top-level ROS CMakeLists.txt
+# However the tests are build separately in the top-level and thus should be ignored here.
+# If you don't know what ROS is, simply ignore this line.
+if(NOT DEFINED ENV{ROS_DISTRO})
```
Comment (martiniil) [inline standalone/CMakeLists.txt:90]:
```suggestion
if(NOT DEFINED ENV{catkin_FOUND})
```
This works also if you have ROS installed.

**Comment by martiniil** [comment]
grr, I don't understand how I destroyed the build ...

**Comment by martiniil** [comment]
@agutenkunst lgtm now

**Comment by agutenkunst** [comment]
@martiniil happy to see the review getting approved and the checkmarks checked.

**Comment by martiniil** [comment]
> 
> 
> @martiniil happy to see the review getting approved and the checkmarks checked.

Should I approve my own changes? I don't want to violate our "4-eyes rule".

**Review by jschleicher — APPROVED** [review]

**Pull request merged**